src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/EndOfServiceReportEventPublisher.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events
+
+import org.springframework.context.ApplicationEvent
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.LocationMapper
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.EndOfServiceReportController
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.EndOfServiceReport
+
+enum class EndOfServiceReportEventType {
+  SUBMITTED,
+}
+
+class EndOfServiceReportEvent(
+  source: Any,
+  val type: EndOfServiceReportEventType,
+  val endOfServiceReport: EndOfServiceReport,
```

Comment:
the EOSR now has a reference to the referral in it, so we can remove the `referralReference` and `ppUser` fields from this even constructor. in the listener class we can pull them out of the referral.